http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -737,32 +726,23 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
 
   private def getLandingZoneResourceName(landingZoneResourcesByPurpose: List[LandingZoneResourcesByPurpose],
                                          resourceType: String,
-                                         purpose: String
-  ): F[String] = {
-    val lzResourcesOpt =
-      landingZoneResourcesByPurpose.find(lzResourceByPurpose => lzResourceByPurpose.purpose.equals(purpose))
-    val resourceOpt = lzResourcesOpt
-      .flatMap(lzResources =>
-        lzResources.deployedResources.find(lzResource => lzResource.resourceType.equalsIgnoreCase(resourceType))
-      )
-    val resourceNameOpt = resourceOpt
-      .map(lzResource =>
-        lzResource.resourceName match {
-          case Some(resourceName) => Some(resourceName)
-          case None =>
-            lzResource.resourceId match {
-              case Some(resourceId) => resourceId.split("/").lastOption
-              case None             => None
-            }
-        }
-      )
-      .flatten
-
-    F.fromOption(
-      resourceNameOpt,
-      AppCreationException(s"${resourceType} resource with purpose ${purpose} not found in landing zone")
-    )
-  }
+                                         purpose: LandingZoneResourcePurpose,
+                                         useParent: Boolean = false
```

Comment:
shall we not have the default? just wanna make sure we wont misuse this at call site